lib/rules/jsx-handler-names.js

```diff
@@ -24,23 +24,56 @@ module.exports = {
       type: 'object',
       properties: {
         eventHandlerPrefix: {
-          type: 'string'
+          oneOf: [
+            {
+              type: 'string'
+            },
+            {
+              type: 'boolean',
+              enum: [false]
+            }
+          ]
         },
         eventHandlerPropPrefix: {
-          type: 'string'
+          oneOf: [
+            {
+              type: 'string'
+            },
+            {
+              type: 'boolean',
+              enum: [false]
+            }
+          ]
         }
       },
       additionalProperties: false
     }]
   },
 
   create(context) {
+    function isPrefixDisabled(prefix) {
+      return prefix === false;
+    }
+
     const configuration = context.options[0] || {};
-    const eventHandlerPrefix = configuration.eventHandlerPrefix || 'handle';
-    const eventHandlerPropPrefix = configuration.eventHandlerPropPrefix || 'on';
 
-    const EVENT_HANDLER_REGEX = new RegExp(`^((props\\.${eventHandlerPropPrefix})|((.*\\.)?${eventHandlerPrefix}))[A-Z].*$`);
-    const PROP_EVENT_HANDLER_REGEX = new RegExp(`^(${eventHandlerPropPrefix}[A-Z].*|ref)$`);
+    const eventHandlerPrefix = isPrefixDisabled(configuration.eventHandlerPrefix) ?
+      null :
+      configuration.eventHandlerPrefix || 'handle';
+    const eventHandlerPropPrefix = isPrefixDisabled(configuration.eventHandlerPropPrefix) ?
+      null :
+      configuration.eventHandlerPropPrefix || 'on';
+
+    if (!eventHandlerPrefix && !eventHandlerPropPrefix) {
+      throw new TypeError('Both \'eventHandlerPrefix\' and \'eventHandlerPropPrefix\' cannot be false');
```

Comment:
i'd also like this to be enforced by the schema, eg one of these three:
1. string / string
1. string / false
1. false / string

Reply:
In the schema definition,
Only allowed  combination are:
1: string / string 
2: string / false
3: false/ string

```javascript
        eventHandlerPrefix: {
          oneOf: [
            {
              type: 'string'
            },
            {
              type: 'boolean',
              enum: [false]
            }
          ]
        },
        eventHandlerPropPrefix: {
          oneOf: [
            {
              type: 'string'
            },
            {
              type: 'boolean',
              enum: [false]
            }
          ]
        }
```

Reply:
right, i'm suggesting we change the schema so that's not possible.

instead of two schemas, each with two variants (total 4), three schemas (total 3).

Reply:
So, Is this ok?
```javascript
    schema: [{
      oneOf: [
        {
          type: 'object',
          properties: {
            eventHandlerPrefix: {type: 'string'},
            eventHandlerPropPrefix: {type: 'string'}
          },
          additionalProperties: false
        }, {
          type: 'object',
          properties: {
            eventHandlerPrefix: {type: 'string'},
            eventHandlerPropPrefix: {
              type: 'boolean',
              enum: [false]
            }
          },
          additionalProperties: false
        }, {
          type: 'object',
          properties: {
            eventHandlerPrefix: {
              type: 'boolean',
              enum: [false]
            },
            eventHandlerPropPrefix: {type: 'string'}
          },
          additionalProperties: false
        }
      ]
    }]
```

Reply:
That looks great!

Reply:
Done

Reply:
that means that this TypeError can never happen, so it can be removed?

Reply:
Oops, True.

Reply:
Done